packages/permissions/src/utils/has-permissions.ts

```diff
@@ -141,3 +172,71 @@ export const getInvalidPermissions = (
     isNil(actualPermissions[toCanCase(demandedPermission)])
   );
 };
+
+const hasDemandedDataFenceForStore = (
+  resourceToApplyDataFence: TResourceToApplyDataFence,
+  options: {
+    actualDataFence: TDataFenceByPermissionName;
+    actualDataFenceName: string;
+    demandedDataFence: TDemandedDataFence;
+  }
+): boolean => {
+  const hasDemandedPermission = hasPermission(options.demandedDataFence.name, {
+    [options.actualDataFenceName]: true,
+  });
+  if (!hasDemandedPermission) return false;
+  switch (options.demandedDataFence.group) {
+    case 'orders': {
+      return options.actualDataFence.values.includes(
+        resourceToApplyDataFence.storeRef &&
+          resourceToApplyDataFence.storeRef.key
+          ? resourceToApplyDataFence.storeRef.key
+          : ''
+      );
+    }
+    default:
+      return false;
```

Comment:
Should we log, throw, use Sentry or anything in this case?

Reply:
IMO we should throw

Reply:
Throwing unmounts the whole app and results in a ice cream. We could throw in dev mode but log to sentry in prod mode.

Reply:
yes they are different. the `demanded*` can be `ManageOrders` whereas the `actual*` can be `ViewOrders`